manifests/init.pp

```diff
@@ -52,108 +52,102 @@
 # * Trevor Vaughan <tvaughan@onyxpoint.com>
 #
 class clamav (
-  $enable_clamav         = defined('$::enable_clamav') ? { true => $::enable_clamav, default => hiera('enable_clamav',true) },
-  $manage_group_and_user = true,
-  $clamav_user           = 'clam',
-  $clamav_group          = 'clam',
-  $package_name          = 'clamav',
-  $enable_freshclam      = false,
-  $schedule_scan         = true,
-  $rsync_source          = "clamav_${::environment}/",
-  $rsync_server          = hiera('rsync::server',''),
-  $rsync_timeout         = hiera('rsync::timeout', '2')
+  Boolean                  $enable                = true,
+  Boolean                  $manage_group_and_user = true,
+  String                   $clamav_user           = 'clam',
+  String                   $clamav_group          = 'clam',
+  String                   $package_name          = 'clamav',
+  Boolean                  $enable_freshclam      = false,
+  Boolean                  $schedule_scan         = true,
+  String                   $rsync_source          = "clamav_${::environment}/",
+  String                   $rsync_server          = simplib::lookup('simp_options::rsync::server', { 'default_value'  => '127.0.0.1' }),
+  Stdlib::Compat::Integer  $rsync_timeout         = simplib::lookup('simp_options::rsync::timeout', { 'default_value' => '2' }),
 ) {
 
-  # Validation.
-  validate_bool($enable_clamav)
-  validate_bool($manage_group_and_user)
-  validate_string($package_name)
-  validate_bool($enable_freshclam)
-  validate_bool($schedule_scan)
-  validate_integer($rsync_timeout)
+  # If the catalyst is disabled, don't manage anything
+  if simplib::lookup('simp_options::clamav', { 'default_value' =>  true }) {
```

Comment:
This module is an implementation of a catalyst, it should not disable itself by default.